drivers/ntc/ntc_tcp.c

```diff
@@ -535,7 +536,10 @@ static int ntc_tcp_server(void *ctx)
 			      SOCK_STREAM, IPPROTO_TCP, &listen_sock);
 	if (rc)
 		goto err_sock;
-
+  rc = kernel_setsockopt(listen_sock, IPPROTO_TCP, TCP_NODELAY, (void*)&flag, sizeof(int));
```

Comment:
We should follow the example of existing kernel code that calls kernel_setsockopt.  Cast the flag to `char *`, and use the variable name in `sizeof(flag)` not the type name.